lib/fetchContent.js

```diff
@@ -323,11 +325,137 @@ function getRecentArticles(days, aspects, facets)
 		});
 }
 
+function getArticleTopics(days, facet, aspects, facets)
+{
+	const date = time.getDatetimeRange('days', days, 0);
+	const params = {
+		queryString: `lastPublishDateTime:> ${date.next}`,
+		maxResults : 100,
+		aspects : aspects,
+		facets : {
+			"names": facets,
+			"maxElements":-1
+		}
+	};
+
+	return searchDeeper(params, 10)
+		.then(results =>  extractArticles(results) )
+		.then(articles => sortArticlesIntoFacets(articles, facet) )
+		.then(topicArticle => wrapResults(topicArticle, days, facet) )
+		.catch(err => {
+			throw err;
+		});
+}
+
+function wrapResults(results, days, facet){
+	return {
+		description: `List of topics, from the last ${days} days. Each topic has a list of articles it features in and a list of other topics listed by those articles`,
+		facet: facet,
+		facetItemTotal: results.facetItemCount,
+		articleTotal: results.articleCount,
+		breakdown: results.breakdown
+	};
+}
+
+
+function sortArticlesIntoFacets(articles, facetName = 'topics')
+{
+	let facetWithArticles = [];
+	let totalArticleCount = 0;
+
+	articles.forEach(article => {
+		let facets = article.metadata[facetName];
+
+		if(!!facets){
+			facets.forEach(facet => {
+
+				let itemObj = objs.getItem(facetWithArticles, 'facetName', facet.term.name);
+
+				if(itemObj === undefined){
+					facetWithArticles.push( newCollection(facet.term.name) );
+					itemObj = objs.getItem(facetWithArticles, 'facetName', facet.term.name);
+				}
+
+				itemObj.articleCount = itemObj.articleCount++;
+				itemObj.articles.push(article);
+
+
+				// related topics
+				let articlesTopics = article.metadata.topics;
```

Comment:
can `articlesTopics` be a `const`? Similarly `articlesPeople` & `articlesOrgs`

Reply:
It feels like those 3 loops could be 1 function with arguments `type` (people, orgs, topics) and  a `content` array

Reply:
Yes, I attempted something earlier but it didn't work correctly so I reverted and will try again later

Reply:
I've amended the variable declarations